src/gc/unix/gcenv.unix.cpp

```diff
@@ -471,13 +488,13 @@ void GCToOSInterface::GetMemoryStatus(uint32_t* memory_load, uint64_t* available
 
         uint64_t available = 0;
         uint32_t load = 0;
+        size_t used;
 
         // Get the physical memory in use - from it, we can get the physical memory available.
         // We do this only when we have the total physical memory available.
-        if (total > 0)
+        if (total > 0 && GetWorkingSetSize(&used))
         {
-            available = sysconf(_SC_PHYS_PAGES) * sysconf(_SC_PAGE_SIZE);
-            uint64_t used = total - available;
+            available = total > used ? total-used : 0; 
             load = (uint32_t)((used * 100) / total);
```

Comment:
This could overflow on 32 bit hardware, you'll need to keep the used type to be uint64_t.